src/Akka.Persistence.MongoDb/Journal/MongoDbJournal.cs

```diff
@@ -302,7 +302,9 @@ protected override Task DeleteMessagesToAsync(string persistenceId, long toSeque
 
         private JournalEntry ToJournalEntry(IPersistentRepresentation message)
         {
+            var timeStamp = DateTimeOffset.Now.ToUnixTimeMilliseconds();
             object payload = message.Payload;
+            message = message.WithTimestamp(timeStamp);
```

Comment:
You can get rid of this call - it's not needed. We already have the equivalent of this with the `Ordering` field - just that in that instance the clock time is set by MongoDb